src/main/java/org/opentripplanner/service/vehiclerental/street/StreetVehicleRentalLink.java

```diff
@@ -53,7 +53,10 @@ public State[] traverse(State s0) {
     }
 
     var preferences = s0.getPreferences().rental(s0.getRequest().mode());
-    if (vehicleRentalPlaceVertex.getStation().networkIsNotAllowed(preferences)) {
+    // preferences will be null while finding nearest places with WALK mode
+    if (
+      preferences != null && vehicleRentalPlaceVertex.getStation().networkIsNotAllowed(preferences)
```

Comment:
Was this a bug even before?

Reply:
Previously we only had generic rental preferences. In a pr before this, I split the rental preferences to be mode specific but there I used bicycle rental preferences as the fallback. Here I removed that fallback which caused the preferences to be null in some cases (as far as I know the nearest search is the only one where you actually need to rental stations without using a rental mode).

This leads to the question in https://github.com/opentripplanner/OpenTripPlanner/pull/5632#issuecomment-1903671563 . I haven't tested or checked the code if we previously used the default preferences from router config in the nearest search as well or if we used some global default settings which can't have banned networks.